src/frontend/screens/Settings/ProjectSettings/DeviceName/index.tsx

```diff
@@ -0,0 +1,223 @@
+          </IconButton>
+        ),
+      });
+    },
+    [navigation],
+  );
+
+  return (
+    <FieldRow label={t(m.deviceNameDisplayHeader)}>
+      <Text>{deviceName}</Text>
+    </FieldRow>
+  );
+}
+
+DeviceName.navTitle = m.title;
+
+function EditMode({
+  deviceName,
+  navigation,
+}: {
+  deviceName: string;
+  navigation: React.ComponentProps<
+    NativeNavigationComponent<'DeviceName'>
+  >['navigation'];
+}) {
+  const {formatMessage: t} = useIntl();
+  const {mutate: updateDeviceName} = useEditDeviceInfo();
+
+  const {control, getValues, handleSubmit} = useForm<{
+    deviceName: string;
+  }>({
+    defaultValues: {deviceName},
+  });
+
+  React.useEffect(
+    function showDiscardChangesAlert() {
+      const unsubscribe = navigation.addListener('beforeRemove', event => {
+        event.preventDefault();
+
+        const nameHasChanges = getValues('deviceName') !== deviceName;
+
+        if (nameHasChanges) {
+          Alert.alert(t(m.discardChangesAlertTitle), undefined, [
+            {style: 'cancel', text: t(m.discardChangesAlertCancelText)},
+            {
+              style: 'default',
+              text: t(m.discardChangesAlertConfirmText),
+              onPress: () => {
+                // TODO: Is going to display mode desired behavior here? Or should it go back to previous screen
+                navigation.setParams({mode: 'display'});
+              },
+            },
+          ]);
+        } else {
+          // TODO: Is going to display mode desired behavior here? Or should it go back to previous screen
+          navigation.setParams({mode: 'display'});
+        }
+      });
+
+      return () => {
+        unsubscribe();
+      };
+    },
+    [t, navigation, getValues, deviceName],
+  );
+
+  React.useEffect(
+    function updateNavigationOptions() {
+      navigation.setOptions({
+        headerRight: () => {
+          return (
+            <IconButton
+              onPress={handleSubmit(value => {
+                updateDeviceName(value.deviceName, {
+                  onSuccess: () => navigation.setParams({mode: 'display'}),
+                  onError: _err => {
+                    // TODO: Handle errors
+                  },
+                });
+              })}>
+              <SaveIcon />
+            </IconButton>
+          );
+        },
+      });
+    },
+    [handleSubmit, navigation, updateDeviceName],
+  );
+
+  return (
+    <FieldRow label={t(m.deviceNameEditHeader)}>
+      <HookFormTextInput
+        control={control}
+        name="deviceName"
+        rules={{maxLength: 100, required: true, minLength: 1}}
+        // TODO: Update HookFormTextInput implementation so that either:
+        // - the implementation fully determines the text input's base style and this component doesn't allow custom styling
+        // - this style prop is properly merged with the text input's base style in the implementation
+        style={{flex: 1, color: BLACK, fontSize: 16}}
```

Comment:
highlighting as it's something i noticed as I was working on this. didn't feel necessary to include such changes in this PR so would defer to a follow-up if desired